core/eval.go

```diff
@@ -261,6 +261,11 @@ func evalGET(args []string, store *Store) []byte {
 		return RespNIL
 	}
 
+	// If the object exists, check if it is a set object.
+	if err := assertType(obj.TypeEncoding, ObjTypeSet); err == nil {
+		return diceerrors.NewErrSetType()
```

Comment:
Please remove this function as encoding is to be done as part of `toEncodedMessage()`.
`WrongTypeErr` is already used at multiple places, we can follow the same pattern.
Ref: `return diceerrors.NewErrWithMessage(diceerrors.WrongTypeErr)`

Reply:
Yes, there are multiple instances for this return, Would you prefer me to fix all of these returns and remove the NewErrSetType()?

Reply:
I think we should follow one pattern across the codebase. And implementation of `NewErrSetType` seems redundant in this case as we already have similar encoding method available and would make it harder from maintainability point also.
So yes we shall change it as `diceerrors.NewErrWithMessage(diceerrors.WrongTypeErr)` if possible.